.github/workflows/go.yml

```diff
@@ -47,12 +51,50 @@ jobs:
           go-version: 1.18
         id: go
       - name: Check out code into the Go module directory
-        uses: actions/checkout@v1
+        uses: actions/checkout@v3
+        with:
```

Comment:
Why do we need this now, when we didn't need it before?  Is it because we've moved to checkout v3?

Reply:
I was trying to follow how kops does it in https://github.com/kubernetes/kops/blob/master/.github/workflows/main.yml as you suggested. It uses a hash but it's checkout v3.
I tried it with checkout v1 and run into an issue. I switched to checkout v3, it works fine.
I didn't spend much time trying to figure out the difference v1 and v3. Do we prefer to use checkout v1?